### ISSUE TITLE
SerialPort config how..?

### ISSUE THREAD

**Issue by sfkristof** [johnlauer/serial-port-json-server#53]
Hi!

What can I do if I would like this tool with a device where I have to use such a serial port config: baud=9600, databits=7, stopbits=1, parity=even, flow control=none?
I couldn't find any possibility to set these and without this the communication is not working.
I tried to use it with a serial port proxy program where I could config these and then it was working.

Thanks in advance!

Kristof

**Comment by chilipeppr** [comment]
You would have to modify the source code to add these extra control parameters into the incoming commands as well as into the code that actually opens the serial port. Then you could do a pull request to add those changes into the mainstream. It's odd that after SPJS being out for over 3 years this question has only come up now. I think that most USB serial ports never care about databits, stopbits, and parity. Are you totally sure your implementation needs this?

**Comment by sfkristof** [comment]
Yes, this is exactly what I did and now it's working!
I modified the OPEN command to manage these like this:
open <port> <baud> <dataBits> <parity> <stopBits> <bufferType>
I don't know why this device uses such a config but it does.
About the forking: I also had to modify this import as a dependency to the SPJS:
github.com\johnlauer\goserial
How should I fork both projects and indicate that the SPJS has a dependency to another fork?

**Comment by chilipeppr** [comment]
I can manually tweak stuff to make the dual dependency work. If you want to do a pull request to send back your changes to me that would be great. I'll try to include them in the main build. Any chance you could tweak the Readme as well with docs on how to send the settings in the open command?

**Comment by sfkristof** [comment]
Of course! Cool, where should I send my version to? (All sources in a ZIP with a txt about the changes is OK?)

**Comment by johnlauer** [comment]
Oh, no, you should do it as a pull request in Github.

**Comment by sfkristof** [comment]
I forked both projects and created 2 pull requests, please tell me if there is still anything to do with them.
